src/Symfony/Component/Mailer/EventListener/EnvelopeListener.php

```diff
@@ -57,7 +62,27 @@ public function onMessage(MessageEvent $event): void
         }
 
         if ($this->recipients) {
-            $event->getEnvelope()->setRecipients($this->recipients);
+            $recipients = $this->recipients;
+            if ($this->allowedRecipients) {
+                foreach ($event->getEnvelope()->getRecipients() as $recipient) {
+                    foreach ($this->allowedRecipients as $allowedRecipient) {
+                        if (!preg_match($allowedRecipient, $recipient->getAddress())) {
```

Comment:
I'm wondering if we shouldn't anchor regexps, so that the config is less error prone and easier to write:
```suggestion
                        if (!preg_match('{'.$allowedRecipient.'}A', $recipient->getAddress())) {
```

this would allow listing `'.*@example.com` to allow all emails from this domain (instead of `'/@example.com$/` at the moment)

Reply:
pr updated